Packages/Tracking/Interaction Engine/Runtime/Scripts/HandRays/WristShoulderFarFieldHandRay.cs

```diff
@@ -0,0 +1,198 @@
+ * Copyright (C) Ultraleap, Inc. 2011-2022.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System;
+using UnityEngine;
+
+namespace Leap.Unity.Interaction
+{
+#pragma warning disable 0618
+
+    /// <summary>
+    /// Calculates a far field hand ray based on the wrist, shoulder and pinch position.
+    /// This allows a developer to decide how responsive the ray feels to the hand's rotation.
+    /// For a more responsive ray, blend the WristShoulderBlendAmount towards the wrist, 
+    /// For a more stable ray, blend the WristShoulderBlendAmount towards the shoulder.
+    /// </summary>
+    public class WristShoulderFarFieldHandRay : HandRay
+    {
+        /// <summary>
+        /// The wrist shoulder lerp amount is only used when the rayOrigin is wristShoulderLerp. 
+        /// It specifies how much the wrist vs the shoulder is used as a ray origin.
+        /// At 0, only the wrist position and rotation are taken into account.
+        /// At 1, only the shoulder position and rotation are taken into account.
+        /// For a more responsive far field ray, blend towards the wrist. For a more stable far field ray,
+        /// blend towards the shoulder. Keep the value central for a blend between the two.
+        /// </summary>
+        [Tooltip("Our far field ray is a direction from a wristShoulder blend position, through a stable pinch position.\n" +
+            "WristShoulderBlendAmount determines the wristShoulder blend position.\n" +
+            " - At 0, only an wrist position is taken into account.\n" +
+            " - At 1, only the shoulder position is taken into account.\n" +
+            " - For a more responsive far field ray, blend towards the wrist.\n" +
+            " - For a more stable far field ray, blend towards the shoulder.\n" +
+            " - Keep the value central for a blend between the two.")]
+        [Range(0f, 1)] public float wristShoulderBlendAmount = 0.532f;
+
+        [SerializeField] private InferredBodyPositions inferredBodyPositions;
+
+        [Header("Debug Gizmos")]
+        [SerializeField] private bool drawDebugGizmos;
+
+        [SerializeField] private bool drawRay = true;
+        [SerializeField] private Color rayColor = Color.green;
+
+        [SerializeField] private bool drawRayAimAndOrigin = true;
+        [SerializeField] private Color rayAimAndOriginColor = Color.red;
+
+        [SerializeField] private bool drawWristShoulderBlend = false;
+        [SerializeField] private Color wristShoulderBlendColor = Color.blue;
```

Comment:
Can these be hidden behind an editor script that uses `specifyConditionalDrawing`?

e.g. specifyConditionalDrawing("drawDebugGizmos", "drawRay");

Reply:
Yeah good shout - I just didnt do this because I was feeling lazy 😆 will add some custom editors in now for this & inferred body positions

Reply:
https://user-images.githubusercontent.com/17143693/179688153-e4221fad-779b-433c-b072-f8111f45f4c4.mp4

tada